### PR TITLE
Angular 12 upgrade

### LINKED ISSUE #78: IE11 Polyfills
There are a couple of instances where we use Array.includes in the components, consumers who need to support IE11 will need to manually polyfill this. We should either mention this in the docs or modify the code. Given the small number of instances it might be worth just modifying the code.

The test app is quite a good place to work on any IE11 issues `npm run test:app`

### LINKED ISSUE #563: Remove support for IE11
We should remove support for IE11 within the codebase - e.g review browserslist, polyfills, css vars package, etc.

See also issue #78 as related to IE11.

This change should go into the `next` branch.

### PR DESCRIPTION
**Pull request by elenagarrone**
# Description

Upgrade Angular to `v12`:
- update Storybook to `v6.5.7`
- enable Storybook to use Webback 5 and update configuration accordingly
- enable [file system caching](https://storybook.js.org/docs/react/builders/webpack#filesystem-caching) for Storybook
- move `defaultValue` in stories within the `argTypes table` as having it outside is deprecated
- add `ng` scripts for starting and building Storybook which allows to add global styles
- refactor the `npm` scripts
- update `pull-request.yml` and deployment script to use the new scripts
- enable `ivy`, remove `ngcc` and fix import issues
- replace [webpack-fix-style-only-entries](https://github.com/fqborges/webpack-fix-style-only-entries) with the suggested [webpack-remove-empty-scripts](https://github.com/webdiscus/webpack-remove-empty-scripts) which is compatible with Webpack 5
- update [browser support](https://github.com/elenagarrone/canopy/blob/d8ae007f90d9e72e6120325c1ed0f4a22c4ff13b/docs/USAGE.md#browser-support) and remove support for IE11 as end of life will be on the 15th of June 2022. In addition to that Angular 12 deprecates it and support will be completely removed in Angular 13 ([test run replicating master](https://github.com/Legal-and-General/canopy/actions/runs/2447413696))
- with the recent `angular.json` script changes the `build-storybook` script requires the `dist` to have been built. For this reason the `master-push.yml` and the `pages-deploy.yml` workflows have been refactored and merged together
- building of the test app, linting and testing stages have been removed from `push_master.yml` as all the checks already run on the pull request workflow (which is up to date with `master`).

**Note:** there is a known bug with the ordering of the stories but I've notified the Storybook team. 

Example of the buttons stories:
Screenshot 2022-06-06 at 14 04 43

While it should be:
```
Buttons
  Primary dark
  Primary light
  Secondary dark
  Secondary light
  Icon only
  Icon with text
```

Resolves #563
Resolves #78 
Relates to #698

# Checklist:

- [x] The commit messages follow the [convention for this project](./blob/master/docs/CONTRIBUTING.md#conventional-commits)
- [x] I have provided an adequate amount of test coverage
- [ ] I have added the functionality to the [test app](./blob/master/docs/CONTRIBUTING.md#build-test-application)
- [ ] I have provided a story in storybook to document the changes
- [x] I have provided documentation in the notes section of the story
- [ ] I have added any new public feature modules to public-api.ts

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -62,7 +85,13 @@
             "styles": [
               "dist/canopy/canopy.css"
             ],
-            "scripts": []
```
Comment (elenagarrone) [inline angular.json:65]: These bits below have been automatically added during the migration.

File: projects/canopy-test-app/src/polyfills.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@
 /** IE10 and IE11 requires the following for NgClass support on SVG elements */
 // import 'classlist.js';  // Run `npm install --save classlist.js`.
 
-import 'core-js/es7/array';
+// import 'core-js/es7/array';
```
Comment (pauleustice) [inline projects/canopy-test-app/src/polyfills.ts:24]: Could this and the classlist line be removed at all?

Comment (elenagarrone) [reply projects/canopy-test-app/src/polyfills.ts]: Updated 👍

**Review by pauleustice — APPROVED** [review]
Excellent work, nicely done! 🎉

**Pull request merged**